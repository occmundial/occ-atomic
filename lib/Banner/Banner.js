import React from 'react';
import PropTypes from 'prop-types';
import grid from '../subatomic/grid';
import Tip from '../Tip';

const Banner = ({
  id, style, className, children, onClose, winWidth, cta, theme = 'promote'
}) => {
  return (
    <Tip
      id={id}
      style={style}
      className={className}
      onClose={onClose}
      size={winWidth < grid.xs ? 'small' : 'large'}
      center
      type="banner"
      cta={cta}
      theme={theme}
      icon
    >{children}</Tip>
  );
};

Banner.propTypes = {
  winWidth: PropTypes.number,
  /** Text of the tip. If you put text between two * characters that text will be displayed as bold. */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  /** Icon name to show on the left. */
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  onClose: PropTypes.func,
  cta: PropTypes.shape({
		text: PropTypes.string,
		href: PropTypes.string,
		target: PropTypes.string,
		onClick: PropTypes.func
	}),
  theme: PropTypes.oneOf(['info', 'warning', 'success', 'error', 'promote']),
};

export default Banner;
