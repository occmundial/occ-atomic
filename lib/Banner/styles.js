import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
