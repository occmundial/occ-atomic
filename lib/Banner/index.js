import injectSheet from 'react-jss';
import Banner from './Banner';
import styles from './styles';

export default injectSheet(styles)(Banner);
