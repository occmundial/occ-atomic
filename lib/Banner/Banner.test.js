import React from 'react';
import { shallow } from 'enzyme';
import Banner from './Banner';
import BannerJSS from './index';

describe("Banner", () => {
    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Banner 
                onClose={() => {}}
                cta={{
                    text: 'Lorem ipsum.',
                    onClick: () => alert('CTA clicked!')
                }}
                theme="info"
            >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
                tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
                veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
                commodo consequat
            </Banner>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Has content centered', () => {
        const wrapper = shallow(
            <Banner 
                onClose={() => {}}
                cta={{
                    text: 'Lorem ipsum.',
                    onClick: () => alert('CTA clicked!')
                }}
                theme="info"
            >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
                tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
                veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
                commodo consequat
            </Banner>
        );
        expect(wrapper.prop('banner')).toEqual(true);
    });
});

describe("BannerJSS", () => {
    it('matches the snapshot', () => {
        const wrapper = shallow(
            <BannerJSS 
                onClose={() => {}}
                cta={{
                    text: 'Lorem ipsum.',
                    onClick: () => alert('CTA clicked!')
                }}
                theme="info"
            >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
                tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
                veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
                commodo consequat
            </BannerJSS>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

