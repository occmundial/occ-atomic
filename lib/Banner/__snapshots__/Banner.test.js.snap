// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WindowSize
    classes={
      Object {
        "container": "container",
        "icon": "icon",
        "iconFlex": "iconFlex",
        "textBanner": "textBanner",
        "textFlex": "textFlex",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "container": "container",
        "icon": "icon",
        "iconFlex": "iconFlex",
        "textBanner": "textBanner",
        "textFlex": "textFlex",
      },
      "size": "md",
      "winWidth": 1024,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "container": "container",
          "icon": "icon",
          "iconFlex": "iconFlex",
          "textBanner": "textBanner",
          "textFlex": "textFlex",
        },
        "size": "md",
        "winWidth": 1024,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Banner styles matches the snapshot 1`] = `
Object {
  "container": Object {
    "backgroundColor": "#222222",
    "padding": 16,
  },
  "icon": Object {
    "cursor": "pointer",
  },
  "iconFlex": Object {
    "@media screen and (min-width:576px)": Object {
      "padding": 0,
    },
    "paddingLeft": 16,
    "paddingTop": 16,
  },
  "textBanner": Object {
    "display": "inline-block",
  },
  "textFlex": Object {
    "@media screen and (min-width:576px)": Object {
      "textAlign": "center",
    },
    "textAlign": "start",
  },
}
`;

exports[`BannerJSS matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(WindowSize) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "container": "WindowSize-container-0-1-1",
        "icon": "WindowSize-icon-0-1-5",
        "iconFlex": "WindowSize-iconFlex-0-1-4",
        "textBanner": "WindowSize-textBanner-0-1-2",
        "textFlex": "WindowSize-textFlex-0-1-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "container": "WindowSize-container-0-1-1",
          "icon": "WindowSize-icon-0-1-5",
          "iconFlex": "WindowSize-iconFlex-0-1-4",
          "textBanner": "WindowSize-textBanner-0-1-2",
          "textFlex": "WindowSize-textFlex-0-1-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
